exposee-app/Routes/users.js

```diff
@@ -44,46 +47,51 @@ router.get('/:user_id/wallet', validate_Token, async (req, res) => {
     return res.status(500).json({ error: 'Internal server error' });
   }
 });
-router.post('/gift', validate_Token, async (req, res) => {
+router.post("/gift", validate_Token, async (req, res) => {
+  const t = await sequelize.transaction();
+
   try {
     const sender_id = req.userId;
     const { receiver_id, amount, video_id } = req.body;
 
     // Check if both sender and receiver exist in the database
-    const senderWallet = await Wallet.findOne({ where: { user_id:sender_id } });
-    console.log('hey there',senderWallet);
-    const receiverWallet = await Wallet.findOne({ where: { user_id: receiver_id } });
-    console.log('hey there 2', receiverWallet);
+    const senderWallet = await Wallet.findOne({ where: { user_id: sender_id }, transaction: t });
+    const receiverWallet = await Wallet.findOne({ where: { user_id: receiver_id }, transaction: t });
 
     if (!senderWallet || !receiverWallet) {
-      return res.status(404).json({ error: 'Sender or receiver not found' });
+      await t.rollback();
+      return res.status(404).json({ error: "Sender or receiver not found" });
     }
 
     // Check if the sender has sufficient balance
     if (senderWallet.amount < amount) {
-      return res.status(400).json({ error: 'Insufficient balance' });
+      await t.rollback();
```

Comment:
similarly if there's insufficient balance, nothing should happen, you don't need to rollback